rclcpp/include/rclcpp/parameter.hpp

```diff
@@ -33,7 +34,15 @@ class Parameter;
 
 namespace node_interfaces
 {
-struct ParameterInfo;
+// Internal struct for holding useful info about parameters
+struct ParameterInfo
+{
+  /// Current value of the parameter.
+  rclcpp::ParameterValue value;
+
+  /// A description of the parameter
+  rcl_interfaces::msg::ParameterDescriptor descriptor;
+};
```

Comment:
@bpwilcox meta: hmm, why not pushing this `descriptor` attribute into `rclcpp::Parameter`? Then we can use `rclcpp::Parameter` everywhere.

Reply:
I do agree that you could put the descriptor in the `rclcpp::Parameter` object, however that seems to be a more far-reaching change than this PR, especially since many user-code uses  `rclcpp::Parameter`. I think it is a good candidate for a follow-up PR.